src/data/api-service.js

```diff
@@ -1,8 +1,20 @@
 import axios from 'axios'
 import { warn } from '~/utils/warn'
+import { AUDIO, IMAGE } from '~/constants/media'
 
 const DEFAULT_REQUEST_TIMEOUT = 30000
 
+/**
+ * Returns a slug with trailing slash for a given resource name.
+ * For media types, converts the name into resource slug when necessary (i.e. pluralizes 'image'),
+ * for other resources uses the resource name as the slug.
+ * @param {import('../constants/media').MediaType|string} resource
+ * @returns {*}
```

Comment:
The `*` (existential type) is basically `any` and will swallow the actual return type, which here is always `string`. Also, something to note, doing `AnyType | AWiderType` will always reduce to just `AWiderType`, so the `param` annotation will just reduce to `string`, which isn't as helpful. We can either leave out the `MediaType` annotation or update the call sites of this function to specify that the input is always `MediaType` (for example update type of `resource` on the `query` function to be `MediaType`).

```suggestion
 * @param {import('../constants/media').MediaType} resource
 * @returns {string}
```